src/web/releases.rs

```diff
@@ -463,50 +450,54 @@ pub fn releases_failures_by_stars_handler(req: &mut Request) -> IronResult<Respo
 pub fn author_handler(req: &mut Request) -> IronResult<Response> {
     let router = extension!(req, Router);
     // page number of releases
-    let page_number: i64 = router.find("page").unwrap_or("1").parse().unwrap_or(1);
-
-    let conn = extension!(req, Pool).get()?;
-
-    #[allow(clippy::or_fun_call)]
+    let page_number: i64 = router
+        .find("page")
+        .and_then(|page_num| page_num.parse().ok())
+        .unwrap_or(1);
     let author = ctry!(router
         .find("author")
-        .ok_or(IronError::new(Nope::CrateNotFound, status::NotFound)));
+        // TODO: Accurate error here, the author wasn't provided
+        .ok_or_else(|| IronError::new(Nope::CrateNotFound, status::NotFound)));
 
-    let (author_name, packages) = if author.starts_with('@') {
-        let mut author = author.split('@');
+    let (author_name, releases) = {
+        let conn = extension!(req, Pool).get()?;
 
-        get_releases_by_owner(
-            &conn,
-            page_number,
-            RELEASES_IN_RELEASES,
-            cexpect!(author.nth(1)),
-        )
-    } else {
-        get_releases_by_author(&conn, page_number, RELEASES_IN_RELEASES, author)
+        if author.starts_with('@') {
+            let mut author = author.split('@');
+
+            get_releases_by_owner(
+                &conn,
+                page_number,
+                RELEASES_IN_RELEASES,
+                // TODO: Is this fallible?
+                cexpect!(author.nth(1)),
+            )
+        } else {
+            get_releases_by_author(&conn, page_number, RELEASES_IN_RELEASES, author)
+        }
     };
 
-    if packages.is_empty() {
+    if releases.is_empty() {
+        // TODO: Accurate error here, the author wasn't found
         return Err(IronError::new(Nope::CrateNotFound, status::NotFound));
     }
 
     // Show next and previous page buttons
-    // This is a temporary solution to avoid expensive COUNT(*)
     let (show_next_page, show_previous_page) = (
-        packages.len() == RELEASES_IN_RELEASES as usize,
+        releases.len() == RELEASES_IN_RELEASES as usize,
         page_number != 1,
     );
```

Comment:
Why did this comment go away? Do you know why it was there originally?

Reply:
Same as the other occurrence